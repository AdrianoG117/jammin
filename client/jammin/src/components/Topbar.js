import React from 'react'

function Topbar() {
  return (
    <div>
      <ul className="navbar">
        <li className="nav-el">Login</li>
        <li className="nav-el">Sign up</li>
      </ul>
    </div>
  )
}

export default Topbar
