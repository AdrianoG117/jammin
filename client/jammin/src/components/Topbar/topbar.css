.nav-container {
  /* background-color: rgb(94, 243, 243); */
  width: 100%;
  height: 7vh;
  display: flex;
  list-style-type: none;
  /* background-color: rgba(0, 0, 0, 0.7); */
}

.nav-container {
  display: flex;
  align-items: center;
  justify-content: center;
}

.navbar-right {
  display: flex;
  margin-left: auto;
}

.nav-el {
  color: white;
  font-size: 1.3rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 1rem;
}

.nav-el:hover {
  cursor: pointer;
  text-decoration: underline;
}

.linktag {
  height: 100%;
  margin: 1rem;
  width: 10%;
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  color: white;
}

.otherlinktag {
  height: 100%;
  margin: 1rem;
  width: 10%;
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  color: white;
  font-size: 1.3rem;
}

.otherlinktag:hover {
  cursor: pointer;
  text-decoration: underline;
}

.nav-container h2 {
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.nav-container h2:hover {
  /* background-color: rgb(147, 105, 245); */
  cursor: pointer;
  text-decoration: underline;
}
