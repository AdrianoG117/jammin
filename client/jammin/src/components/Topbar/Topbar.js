import React from 'react';
import { Link } from 'react-router-dom';
import './topbar.css';
import { useHistory } from 'react-router-dom';

function Topbar() {
  const history = useHistory();

  return (
    <div className="nav-container">
      <Link to="/" class="linktag">
        <h2>Home</h2>
      </Link>
      <Link to="/createjam" class="otherlinktag">
        <li>Create Jam</li>
      </Link>
      <Link to="/findjam" class="otherlinktag">
        <li>Find Jam</li>
      </Link>

      <ul className="navbar-right">
        <li className="nav-el" onClick={() => history.push(`/login`)}>
          Login
        </li>
        <li className="nav-el" onClick={() => history.push(`/signup`)}>
          Sign up
        </li>
      </ul>
    </div>
  );
}

export default Topbar;
