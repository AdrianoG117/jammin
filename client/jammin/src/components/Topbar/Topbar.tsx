import React, { FunctionComponent, SetStateAction } from 'react';
import "./topbar.css";
import { useHistory } from "react-router-dom";
import { User } from "../../apiService/APIResponseTypes";
import { initialUserState } from '../../App'

interface IProps {
  setUserData: React.Dispatch<SetStateAction< User >>; 
  setIsSignedUp: React.Dispatch<SetStateAction<boolean>>;
  isSignedUp: boolean;
}

const Topbar:React.FunctionComponent<IProps> = ({ setUserData, setIsSignedUp, isSignedUp }:IProps) => {
  const history = useHistory();

  return (
    <div className="nav-container">
      <div className="home-tag" onClick={() => history.push("/")} >
        <h2>Home</h2>
      </div>
      <div className="menu-tag" onClick={() => history.push("/dashboard")} >
        <li>Create Jam</li>
      </div>
      <div className="menu-tag" onClick={() => history.push("/dashboard")} >
        <li>Find Jam</li>
      </div>

      {isSignedUp ? (
        <ul className="navbar-right">
          <li className="nav-el" onClick={() => history.push("/dashboard")} >
            Dashboard
          </li>
          <li
            className="nav-el"
            onClick={() => {
              setUserData(initialUserState);
              setIsSignedUp(false);
              history.push("/login");
            }}
          >
            Log Out
          </li>
        </ul>
      ) : (
        <ul className="navbar-right">
          <li className="nav-el" onClick={() => history.push("/login")}>
            Login
          </li>
          <li className="nav-el" onClick={() => history.push("/signup")}>
            Sign up
          </li>
        </ul>
      )}
    </div>
  );
}

export default Topbar;
