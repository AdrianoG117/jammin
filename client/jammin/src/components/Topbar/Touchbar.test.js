import {render, screen} from '@testing-library/react';
import Topbar from './Topbar';


describe('Topbar component', ()=> {
  
  test('should match the snapshot', () => {
    const {container} = render(<Topbar isSignedUp={[]}
                  setUserData={()=>{}}
                  setIsSignedUp={()=>{}} />);
    expect(container.firstChild).toMatchSnapshot();
  })
  
  test('Should render the headings', () => {
    render (<Topbar isSignedUp={[]}
                  setUserData={()=>{}}
                  setIsSignedUp={()=>{}} />);
    //title
    screen.getByText(/Home/)
    screen.getByText(/Create Jam/)
    screen.getByText(/Find Jam/)
  })

  test('Should render Dashboard and Log Out if signed up', () => {
    render (<Topbar isSignedUp={true}
        setUserData={()=>{}}
        setIsSignedUp={()=>{}} />);
    //title
    screen.getByText(/Dashboard/)
    screen.getByText(/Log Out/)
    })

  test('Should render Login and Sign up if not signed up', () => {
    render (<Topbar isSignedUp={false}
        setUserData={()=>{}}
        setIsSignedUp={()=>{}} />);
    //title
    screen.getByText(/Login/)
    screen.getByText(/Sign up/)
    })
})

