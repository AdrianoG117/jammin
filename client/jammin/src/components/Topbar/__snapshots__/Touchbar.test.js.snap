// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar component should match the snapshot 1`] = `
<div
  class="nav-container"
>
  <div
    class="home-tag"
    role="button"
    tabindex="0"
  >
    <h2>
      Home
    </h2>
  </div>
  <div
    class="menu-tag"
  >
    <li
      role="presentation"
    >
      Create Jam
    </li>
  </div>
  <div
    class="menu-tag"
    role="button"
    tabindex="0"
  >
    <li>
      Find Jam
    </li>
  </div>
  <ul
    class="navbar-right"
  >
    <li
      class="nav-el"
      role="presentation"
    >
      Dashboard
    </li>
    <li
      class="nav-el"
      role="presentation"
    >
      Log Out
    </li>
  </ul>
</div>
`;
